pydantic/dataclasses.py

```diff
@@ -30,6 +30,8 @@ def _pydantic_post_init(self: 'DataclassType') -> None:
     d = validate_model(self.__pydantic_model__, self.__dict__, cls=self.__class__)[0]
     object.__setattr__(self, '__dict__', d)
     object.__setattr__(self, '__initialised__', True)
+    if self.__post_init_original__ and hasattr(self, '__post_init_post_parse__'):
```

Comment:
I think you need to store a reference to `__post_init_post_parse__` on self the same way `__post_init_original__` is below, then this would be

```py
if self.__post_init_post_parse__
    self.__post_init_post_parse__()
```

Or maybe I'm missing something?